frontend/src/components/ExternalResourceModal.js

```diff
@@ -0,0 +1,93 @@
+import React, { useCallback, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import {
+  Button, Modal, Alert, useModal, connectModal,
+} from '@trussworks/react-uswds';
+
+import { isValidURL } from '../utils';
+
+const ESCAPE_KEY_CODE = 27;
+
+const ExternalResourceModal = ({ onOpen, onClose }) => (
```

Comment:
For some reason I couldn't get the modal to disappear if the user clicks outside of it. I looked at how context menu was doing it but no success, but escape does work

Reply:
Honestly once we focus on the cancel button when the modal opens we may not even need to close the menu on escape, the file delete modal doesn't close on escape.

Reply:
True. ^ Is closing of the modal something that was requested? We could modify the event listener to include other keys besides the escape key if that were the case.